File: config/src/artifact/vorpal.rs
use crate::artifact::SYSTEMS;
use anyhow::Result;
use vorpal_sdk::{artifact::language::rust::Rust, context::ConfigContext};

#[derive(Default)]
pub struct Vorpal {}

impl Vorpal {
    pub fn new() -> Self {
        Self::default()
    }

    pub async fn build(self, context: &mut ConfigContext) -> Result<String> {
        Rust::new("vorpal", SYSTEMS.to_vec())
            .with_bins(vec!["vorpal"])
            .with_includes(vec!["cli", "sdk/rust"])
            .with_packages(vec!["vorpal-cli", "vorpal-sdk"])
            .build(context)
            .await
    }
}

File: sdk/go/cmd/vorpal/artifact/vorpal.go
package artifact

import (
	"github.com/ALT-F4-LLC/vorpal/sdk/go/pkg/artifact/language"
	"github.com/ALT-F4-LLC/vorpal/sdk/go/pkg/config"
)

func Vorpal(context *config.ConfigContext) (*string, error) {
	return language.NewRust("vorpal", SYSTEMS).
		WithBins([]string{"vorpal"}).
		WithIncludes([]string{"cli", "sdk/rust"}).
		WithPackages([]string{"vorpal-cli", "vorpal-sdk"}).
		Build(context)
}
